evap/static/ts/src/auto-form-saver.ts

```diff
@@ -0,0 +1,198 @@
+/**
+Derived from https://github.com/simsalabim/sisyphus, originally distributed under the following license:
+
+---------------------------------------------- begin original license
+Copyright (c) 2011-2013 Alexander Kaupanin https://github.com/simsalabim
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+---------------------------------------------- end original license
+
+New portions and modifications are licensed under the conditions in LICENSE.md
+ */
+
+interface AutoFormSaverOptions {
```

Comment:
Out of curiosity: Did we consider using `type` instead of `interface`, and did we have any reason for one or the other? I would think that with `Partial` in the constructor, `type` would be the correct construct, but not sure